pyflakes/messages.py

```diff
@@ -11,6 +11,8 @@ def __init__(self, filename, loc):
         self.filename = filename
         self.lineno = loc.lineno
         self.col = getattr(loc, 'col_offset', 0)
+        self.end_col = getattr(loc, 'end_col_offset', None)
+        self.end_lineno = getattr(loc, 'end_lineno', None)
```

Comment:
As I stated in the issue, I think it would much better to make the default to be equal to the start. That way the type is always an integer, and start = end will typically just do the right thing for most code that would deal with these things.

Reply:
I disagree. In the use-case of editors when we want to highlight a range with wavy underline, providing start = end will make the underline invisible. start = end might be handled as a special case e.g. when a character is missing (and then denoted with another type of marker like an arrow pointing to that location in the editor). Coercing all missing data to that representation prevents the editors from distinguishing when they need to show the arrow marker and when they need to infer the underline range themselves (e.g default it to the closest token or the entire line).

Reply:
If editors want to show a different marker, can't they just check if start == end? 

We could also consider this an off by one type issue where we could make the default to be to select a single character.